mediapipe/tasks/c/vision/pose_landmarker/pose_landmarker.h

```diff
@@ -0,0 +1,149 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef MEDIAPIPE_TASKS_C_VISION_POSE_LANDMARKER_POSE_LANDMARKER_H_
+#define MEDIAPIPE_TASKS_C_VISION_POSE_LANDMARKER_POSE_LANDMARKER_H_
+
+#include "mediapipe/tasks/c/core/base_options.h"
+#include "mediapipe/tasks/c/vision/core/common.h"
+#include "mediapipe/tasks/c/vision/pose_landmarker/pose_landmarker_result.h"
+
+#ifndef MP_EXPORT
+#define MP_EXPORT __attribute__((visibility("default")))
+#endif  // MP_EXPORT
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// The options for configuring a MediaPipe pose landmarker task.
+struct PoseLandmarkerOptions {
+  // Base options for configuring MediaPipe Tasks, such as specifying the model
+  // file with metadata, accelerator options, op resolver, etc.
+  struct BaseOptions base_options;
+
+  // The running mode of the task. Default to the image mode.
+  // PoseLandmarker has three running modes:
+  // 1) The image mode for recognizing pose landmarks on single image inputs.
+  // 2) The video mode for recognizing pose landmarks on the decoded frames of a
+  //    video.
+  // 3) The live stream mode for recognizing pose landmarks on the live stream
+  //    of input data, such as from camera. In this mode, the "result_callback"
+  //    below must be specified to receive the detection results asynchronously.
+  RunningMode running_mode;
+
+  // The maximum number of poses can be detected by the PoseLandmarker.
+  int num_poses = 1;
+
+  // The minimum confidence score for the pose detection to be considered
+  // successful.
+  float min_pose_detection_confidence = 0.5;
+
+  // The minimum confidence score of pose presence score in the pose landmark
+  // detection.
+  float min_pose_presence_confidence = 0.5;
+
+  // The minimum confidence score for the pose tracking to be considered
+  // successful.
+  float min_tracking_confidence = 0.5;
+
+  // Whether to output segmentation masks.
+  bool output_segmentation_masks = false;
+
+  // The user-defined result callback for processing live stream data.
+  // The result callback should only be specified when the running mode is set
+  // to RunningMode::LIVE_STREAM. Arguments of the callback function include:
+  // the pointer to recognition result, the image that result was obtained
+  // on, the timestamp relevant to recognition results and pointer to error
+  // message in case of any failure. The validity of the passed arguments is
+  // true for the lifetime of the callback function.
+  //
+  // A caller is responsible for closing pose landmarker result.
+  typedef void (*result_callback_fn)(PoseLandmarkerResult* result,
```

Comment:
This should be a const reference (result).

We also still need to update the other tasks to use the same type.